cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/IndexManagerImpl.java

```diff
@@ -205,9 +207,19 @@ public void close() {
      */
     @Override
     public String ensureIndexed(List<FieldSort> fieldNames) throws QueryException {
+        List<Index> indexes = this.listIndexes();
+        Collections.sort(fieldNames);
+        for(Index index: indexes){
+            Collections.sort(index.fieldNames);
+            if (fieldNames.equals(filterMeta(index.fieldNames))){
+                return index.indexName;
+            }
+        }
+
         return this.ensureIndexed(fieldNames, null);
```

Comment:
Do we need to do all the stuff before this `return` statement in this method? Can't we just do the `return` statement because the checks before this are included in the version of `ensureIndexed` ultimately called?